internal/websocket/ws.go

```diff
@@ -130,8 +169,11 @@ func (ws *WebSocket) initialize() {
 					// TODO need to find a proper way to log this error
 					continue
 				}
-
-				ws.respChan <- res
+				responseChan, ok := ws.getResponseChannel(fmt.Sprintf("%v", res.ID))
+				if ok {
+					responseChan <- res
+					close(responseChan)
+				}
```

Comment:
I think we should have a TODO here to remind us to log in an else clause, because this would definitely not be normal